Army/ArmyLRFull.cs

```diff
@@ -472,106 +399,50 @@ public void ArmyLTs(int quant)
-            }
-        }
-        if (!hasDarkCaster)
-            ILDC.GetILDC(false);
+        if (item == null)
+            return;
 
-        Core.AddDrop("Legion Token");
-        Core.AddDrop(LRMaterials);
-        Core.AddDrop(LF1);
+        Core.AddDrop(item);
 
-        Core.Logger($"Farming {quant} Revenant's Spellscroll");
-        Bot.Quests.UpdateQuest(2060);
-        Core.RegisterQuests(6897);
-        int i = 1;
-        while (!Bot.ShouldExit && !Core.CheckInventory("Revenant's Spellscroll", quant))
-        {
-            Core.EquipClass(ClassType.Solo);
-            Adv.BestGear(GearBoost.Undead);
-            /*Sells non-full stacks to keep in sync for each LF1 quest item*/
-            if (!Core.CheckInventory("Aeacus Empowered", 50))
-                Core.SellItem("Aeacus Empowered", 0, true);
-            ArmyHunt("judgement", new[] {"Ultra Aeacus"}, "Aeacus Empowered", false, 50);
+        Core.EquipClass(classType);
+        Core.FarmingLogger(item, quant);
 
-            Core.EquipClass(ClassType.Farm);
-            Adv.BestGear(GearBoost.dmgAll);
-            
-            if (!Core.CheckInventory("Tethered Soul", 300))
-                Core.SellItem("Tethered Soul", 0 , true);
-            ArmyHunt("revenant",  new[] {"Forgotten Soul"}, "Tethered Soul", false, 300);
-            
-            if (!Core.CheckInventory("Darkened Essence", 500))
-                Core.SellItem("Darkened Essence", 0, true);
-            ArmyHunt("shadowrealmpast", new[] { "Pure Shadowscythe, Shadow Guardian, Shadow Warrior" }, "Darkened Essence", false, 500);
+        Core.Join(map);
+        WaitCheck();
 
-            Bot.Quests.UpdateQuest(2061);
-            Adv.BestGear(GearBoost.Undead);
-            
-            if (!Core.CheckInventory("Dracolich Contract", 1000))
-                Core.SellItem("Dracolich Contract", 0, true);
-            ArmyHunt("necrodungeon", new[] { "5 Headed Dracolich" }, "Dracolich Contract", false, 1000);
+        Army.SmartAggroMonStart(map, monsters);
 
-            Bot.Drops.Pickup("Revenant's Spellscroll");
-            Core.Logger($"Completed x{i++}");
-        }
-        Core.CancelRegisteredQuests();
+        while (!Bot.ShouldExit && !Core.CheckInventory(item, quant))
+            Bot.Combat.Attack("*");
+
+        Army.AggroMonStop(true);
+        Core.JumpWait();
     }
 
-    public void ArmyLF3(int quant)
+    void WaitCheck()
     {
-        Core.Logger($"Farming {quant} Exalted Crown");
-        while (!Bot.ShouldExit && !Core.CheckInventory("Exalted Crown", 10))
+        while (Bot.Map.PlayerCount < Bot.Config.Get<int>("armysize"))
         {
-            Core.RegisterQuests(6899);
-            Core.BuyItem("underworld", 216, "Hooded Legion Cowl");
-            /*This is the only not prefarmed item left to get*/
-            ArmyDarkTokenOfDage(100);
-            if (!Core.CheckInventory("Legion Token", 4000))
-                ArmyLTs(4000);
-            Bot.Drops.Pickup("Exalted Crown");
```

Comment:
just do a Core.AddDrop(if it isnt in one already,) and a Bot.Wait.ForPickup("item"); at the end